src/main/webapp/app/exam/participate/exercises/file-upload/file-upload-exam-submission.component.ts

```diff
@@ -126,7 +130,28 @@ export class FileUploadExamSubmissionComponent extends ExamSubmissionComponent i
         }
     }
 
+    /**
+     *  Here we send the submissionFile obtained in setFileSubmissionForExercise() to the server with the update method. The server returns the path to the file, and we
+     *  set it in the submission.
+     */
     saveUploadedFile() {
-        this.onExerciseChanged.emit({ exercise: this.exercise, force: false });
+        if (!this.submissionFile) {
+            return;
+        }
+        this.fileUploadSubmissionService.update(this.studentSubmission as FileUploadSubmission, this.exercise.id!, this.submissionFile).subscribe(
+            (res) => {
+                const submissionFromServer = res.body!;
+                this.studentSubmission.filePath = submissionFromServer.filePath;
+                this.studentSubmission.isSynced = true;
+                this.studentSubmission.submitted = true;
+                this.updateViewFromSubmission();
+            },
+            () => this.onSaveSubmissionError(),
+        );
+    }
+
+    private onSaveSubmissionError() {
+        // show only one error for 5s - see constructor
+        this.synchronizationAlert$.next();
```

Comment:
I am not really seeing where you are subscribing and unsubscribing (to prevent memory leak) from this Subject. What should I see in the constructor?

Reply:
Its refering to the constructor in the exam-participation.component. I guess the issue is that I just copied the method from there to handle errors somehow. Tbh I don't know how to handle this error in the best way

Reply:
Yes but I see what it does in exam-participation.component but you are not doing anything like this in the constructor
`
        // show only one synchronization error every 5s
        this.synchronizationAlert$.pipe(throttleTime(5000)).subscribe(() => this.alertService.error('artemisApp.examParticipation.saveSubmissionError'));
`
This is waht should be in the constructor

Reply:
Please also do something like this         const subscription = this.synchronizationAlert$.pipe(throttleTime(5000)).subscribe(() => this.alertService.error('artemisApp.examParticipation.saveSubmissionError'));

And **desubscribe the subscription here and in exam-participation.component.ts in the ngDestroy** Otherwiese we have a memory leak

Reply:
https://kgotgit.medium.com/rxjs-subject-subscription-heap-memory-analysis-909dc173a613

Reply:
Thanks for your help :D I changed it. Now the error is handled properly.